Media async: ConcurrentModificationException

```
07-07 15:31:44.561 1698-1698/org.wordpress.android E/AndroidRuntime: FATAL EXCEPTION: main
                                                                     org.greenrobot.eventbus.EventBusException: Invoking subscriber failed
                                                                         at org.greenrobot.eventbus.EventBus.handleSubscriberException(EventBus.java:505)
                                                                         at org.greenrobot.eventbus.EventBus.invokeSubscriber(EventBus.java:487)
                                                                         at org.greenrobot.eventbus.EventBus.invokeSubscriber(EventBus.java:479)
                                                                         at org.greenrobot.eventbus.HandlerPoster.handleMessage(HandlerPoster.java:67)
                                                                         at android.os.Handler.dispatchMessage(Handler.java:99)
                                                                         at android.os.Looper.loop(Looper.java:137)
                                                                         at android.app.ActivityThread.main(ActivityThread.java:5041)
                                                                         at java.lang.reflect.Method.invokeNative(Native Method)
                                                                         at java.lang.reflect.Method.invoke(Method.java:511)
                                                                         at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:793)
                                                                         at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:560)
                                                                         at dalvik.system.NativeStart.main(Native Method)
                                                                      Caused by: java.util.ConcurrentModificationException
                                                                         at java.util.HashMap$HashIterator.nextEntry(HashMap.java:792)
                                                                         at java.util.HashMap$KeyIterator.next(HashMap.java:819)
                                                                         at org.wordpress.android.ui.media.services.MediaUploadService.updatePostAndStopServiceIfUploadsComplete(MediaUploadService.java:330)
                                                                         at org.wordpress.android.ui.media.services.MediaUploadService.uploadNextInQueue(MediaUploadService.java:192)
                                                                         at org.wordpress.android.ui.media.services.MediaUploadService.handleOnMediaUploadedSuccess(MediaUploadService.java:137)
                                                                         at org.wordpress.android.ui.media.services.MediaUploadService.onMediaUploaded(MediaUploadService.java:396)
                                                                         at java.lang.reflect.Method.invokeNative(Native Method)
                                                                         at java.lang.reflect.Method.invoke(Method.java:511)
                                                                         at org.greenrobot.eventbus.EventBus.invokeSubscriber(EventBus.java:485)
                                                                         at org.greenrobot.eventbus.EventBus.invokeSubscriber(EventBus.java:479) 
                                                                         at org.greenrobot.eventbus.HandlerPoster.handleMessage(HandlerPoster.java:67) 
                                                                         at android.os.Handler.dispatchMessage(Handler.java:99) 
                                                                         at android.os.Looper.loop(Looper.java:137) 
                                                                         at android.app.ActivityThread.main(ActivityThread.java:5041) 
                                                                         at java.lang.reflect.Method.invokeNative(Native Method) 
                                                                         at java.lang.reflect.Method.invoke(Method.java:511) 
                                                                         at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:793) 
                                                                         at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:560) 
                                                                         at dalvik.system.NativeStart.main(Native Method) 
07-07 15:31:44.561 380-390/system_process W/ActivityManager:   Force finishing activity org.wordpress.android/.ui.main.WPMainActivity
07-07 15:31:44.641 380-383/system_process D/dalvikvm: GC_CONCURRENT freed 975K, 23% free 13064K/16788K, paused 1ms+2ms, total 9ms
````

##### Tested on [Samsung Galaxy S3], Android [7.1.2]

--- a/WordPress/src/main/java/org/wordpress/android/ui/media/services/MediaUploadService.java
+++ b/WordPress/src/main/java/org/wordpress/android/ui/media/services/MediaUploadService.java
@@ -14,7 +14,6 @@ import org.wordpress.android.fluxc.Dispatcher;
 import org.wordpress.android.fluxc.generated.MediaActionBuilder;
 import org.wordpress.android.fluxc.generated.PostActionBuilder;
 import org.wordpress.android.fluxc.model.MediaModel;
-import org.wordpress.android.fluxc.model.MediaModel.MediaUploadState;
 import org.wordpress.android.fluxc.model.PostModel;
 import org.wordpress.android.fluxc.model.SiteModel;
 import org.wordpress.android.fluxc.store.MediaStore;

--- a/WordPress/src/main/java/org/wordpress/android/ui/media/services/MediaUploadService.java
+++ b/WordPress/src/main/java/org/wordpress/android/ui/media/services/MediaUploadService.java
@@ -36,6 +36,7 @@ import java.util.ArrayList;
 import java.util.HashMap;
 import java.util.List;
 import java.util.Map;
+import java.util.concurrent.ConcurrentHashMap;
 
 import javax.inject.Inject;
 
@@ -53,7 +54,7 @@ public class MediaUploadService extends Service {
 
     // To avoid conflicts by editing the post each time a single upload completes, this map tracks completed media by
     // the post they're attached to, allowing us to update the post with the media URLs in a single batch at the end
-    private static HashMap<Integer, List<MediaModel>> mCompletedUploadsByPost = new HashMap<>();
+    private static ConcurrentHashMap<Integer, List<MediaModel>> mCompletedUploadsByPost = new ConcurrentHashMap<>();
 
     @Inject Dispatcher mDispatcher;
     @Inject MediaStore mMediaStore;

